### DISCUSSION TITLE
useHandle doesn't support custom flow IDs

### DISCUSSION THREAD

**Discussion by heaven-born** [bcakmakoglu/vue-flow#973] [Ideas]
`useHandle` can't be used with custom flow IDs. Would be nice to have this option because this function is useless for app that use multiple flows.

**Comment by bcakmakoglu** [comment]
You'll have to be a bit more descriptive so that I can understand what you actually mean 😄 
What are you trying to do?
What isn't working?
What would your suggestion solve?

**Reply by heaven-born** [reply]
I partially figured it out. I wanted something like `useHandle ( {...}, vueFlowId )`, but looks like it takes this ID from the scope. 
However, I think there can be cases when the ID is still not in the context when `useHandle` is called. Let's say I have two flows on the screen and a button outside both of the flow components. And I want this button to start the connection from the node located on one of the flow components. At that point, there won't be a scope with the ID available.

**Reply by bcakmakoglu** [reply]
You're not supposed to use composables in callback functions.
A composable should, in general, be called inside the setup body of a component.
So if you're trying to do something like this

```ts
function onClick() {
	const handler = useHandle(/* ... */)

	// the rest of your code
}
```

I very much advise you *don't* do that 😄 

Instead do this

```ts
const handler = useHandle(/* ... */)

function onClick() {
	// the rest of your code
}
```

**Reply by heaven-born** [reply]
But let's say I need to do something like this:
```javascript
const handler = useHandle(/* ... */)
function onClick(e) {
   handler.handlePointerDown(e)
}
```

But this code is located outside the tree of the vueflow component. So the template looks like this:

```html
<VueFlow id='first'/> 
<VueFlow id='second'/> 
<button @click='onClick'> Start connection from node in 'first' flow </button>
```

At this point, `useHandle` have no information about the ID of the flow that I'm trying to perform the operation on.

**Reply by bcakmakoglu** [reply]
Yeah, that is right, it's not intended for that sort of usage.
It's intention is to create a custom handle if necessary. This is sort of out of the scope of it's purpose.

Also: What are you trying to achieve with that button? 🤔
You're not gonna be able to use that button to create an edge from it to some node, I can tell you that much.

**Reply by heaven-born** [reply]
I managed to find an alternative approach with a repositioning handle, but the initial idea was to position this button on top of one of my flow components and use it as a replacement for a handle. I don't already remember why I tried to do this way, but there was a reason)
